src/System.Management.Automation/engine/Modules/PSModuleInfo.cs

```diff
@@ -301,49 +301,37 @@ private void SetPSDataPropertiesFromPrivateData()
             ProjectUri = null;
             IconUri = null;
 
-            var privateDataHashTable = _privateData as Hashtable;
-            if (privateDataHashTable != null)
+            if (_privateData is Hashtable hashData && hashData["PSData"] is Hashtable psData)
             {
-                var psData = privateDataHashTable["PSData"] as Hashtable;
-                if (psData != null)
+                var tagsValue = psData["Tags"];
+                if (tagsValue is object[] tags && tags.Length > 0)
                 {
-                    object tagsValue = psData["Tags"];
-                    if (tagsValue != null)
+                    foreach (var tagString in tags.OfType<string>())
```

Comment:
Policy here is that any other object in a "Tags" array that isn't a string will be ignored rather than an error or coerced?

Reply:
That's the logic from the original code, so I guess yes.